machine_management/creating-infrastructure-machinesets.adoc

```diff
@@ -43,6 +43,22 @@ include::modules/machineset-yaml-gcp.adoc[leveloffset=+3]
 
 include::modules/machineset-creating.adoc[leveloffset=+2]
 
+[id="assigning-machineset-resources-to-infra-nodes"]
+== Assigning MachineSet resources to infrastructure nodes
+
+After creating an infrastructure MachineSet, the MachineSet assigns the `worker`
+and `infra` roles to the nodes it manages. The Machine Config Operator (MCO)
```

Comment:
@enxebre is this true? The last time I recall machines did not assign worker/infra roles to nodes. MCO (based on configuration) decided if a node is master/worker.

Reply:
@enxebre bump. Can you confirm the above text? This originates from [BZ#1760675](https://bugzilla.redhat.com/show_bug.cgi?id=1760675#c23). From there, I gathered
>The design of the MCO is such that there is no intrinsic notion of an infra node. As a result, the infra nodes should carry two roles: infra and worker. The worker role is needed by the MCO in order to manage the lifecycle of the machines' configuration and the infra role is just needed for billing purposes.

Thanks!

Reply:
@runcom can you provide any insight here? This relates to my [BZ#1796054 comment](https://bugzilla.redhat.com/show_bug.cgi?id=1796054#c16).

Reply:
From a scheduling perspective taints are probably the best way to ensure a pod doesn't get scheduled onto a node. I'm not sure of the details of the MCO, but I think it just reconciles these labels to the matching taint (don't quote me on that). Might be best to ask the MCO team to confirm